### ISSUE TITLE
(5) Add parent records for the MS20 Collection 

### ISSUE THREAD

**Issue by kellyChex** [UCSCLibrary/dams_project_mgmt#92]
### Steps 
1.  Update the existing metadata
2.  Ingest the parent records  

### Acceptance Criteria 
- Metadata loaded into Fedora correctly

**Comment by NedHenry** [comment]
I think the descriptive summary is a little confusing. Here's my understanding (let me know if anything seems off base):
Right now each photograph in this collection is a work in the DAMS, with full descriptive metadata record and searchability. These photographs fall naturally into sets of images depicting the same subject (usually one photo from each set has been chosen for official publication), so we want to create works corresponding to the sets of images. We are adding parent works that group these existing works together. Each parent work is a set of photographs, and each child work is a photograph. Ideally, both parents and children would be able to show up in search results, and the parent works would show up first by default.

**Comment by NedHenry** [comment]
Just a head up - I think I was a little confused when I was pointing this one, and I may have pointed it a bit too low. I was thinking about just ingesting the parent records. Updating the metadata on the child records may involve a bit of debugging of the BulkOps gem. I just wanted to give a heads up that this might involve more effort than is reflected here.

**Comment by rmjaffe** [comment]
@NedHenry I moved the remodeled MS 20 spreadsheets in dams_ingest: smb://artifacts.library.ucsc.edu/dams_ingest/ms0020_remodel_spreadsheets

**Comment by NedHenry** [comment]
Unfortunately, the speed of this bulk ingest/update has been continually worsening due to some Fedora performance problems (#160). Right now there are still a few hundred works in this collection that have not yet updated. The spreadsheets have been parsed and the updated/create tasks have been queued; right now we are only bottlenecked through this Fedora issue. 

I think at this point it would be helpful to start QA on the series that we think have been updated successfully. If anyone wants to spend some time looking through the Ruth Marion Baruch photographs or Black Panther Party series, that would be very helpful. If these Fedora performance problems that we don't yet understand are associated with any issues in our repository data, we want to know sooner rather than later (I haven't seen anything like that so far, but I haven't been looking very thoroughly either yet).

**Comment by rmjaffe** [comment]
Thanks for the update @NedHenry  -- I can start taking a look at these.  Which collection contains the images needing review?

**Comment by rmjaffe** [comment]
@NedHenry:  I've only had a chance to look at a couple of the Black Panthers image set and for the most part, so far so good!  I did note one issue on the parent record for the set [Kathleen Cleaver speaking at Unitarian Church, San Rafael](http://digitalcollections.library.ucsc.edu/records/7p88cj87m).  The only subjectName should be "Black Panther Party" -- not sure how Baby D and Al Lowry got in there.

This also leads me to ask:  Where/how would these types of issues best be reported?

**Comment by rmjaffe** [comment]
This was a different issue, but in reviewing Black Panthers I love seeing that it's been resolved.  Thank you!

![Screen Shot 2020-01-14 at 10 13 12 AM](https://user-images.githubusercontent.com/32003407/72370306-d09b2e00-36b6-11ea-9ca1-438424c02c5c.png)

**Comment by kellyChex** [comment]
@rmjaffe @JessWaggs @chesleyperry Any issues you find related to the ticket should be reported here. If this ticket was in QA it would move back to In Progress unless we determined the issue was out of the ticket's scope then we would create a new ticket.

**Comment by rmjaffe** [comment]
I put together an [QA/error reporting spreadsheet](https://drive.google.com/a/ucsc.edu/file/d/1KtOY3NFft8urlSHJDe2sldZZ3hV8WIl2/view?usp=sharing) for the Black Panthers.  Given the spot checking I did yesterday and the more systematic review of two image sets I did just not, the metadata looks good:  metadata properties and values are showing up; inheritance is working as expected.  But, I'm seeing an issue where there are lost child images.  I think we will need to talk more about this as, depending on the nature of the issue, it may require more all hands-type, extensive review.

**Comment by rmjaffe** [comment]
Also observing that metadata properties (theme, subseries) that are no longer included in the metadata and were not included re-ingest spreadsheets are still part of the metadata records in the DAMS.

**Comment by NedHenry** [comment]
So far the ingest/update software applies changes only to the columns present in the spreadsheet. It by default ignores metadata elements that are not included in the spreadsheet. 

Once we get the fedora performance problems sorted out, finish implementing all the complex object relationships, and track down whatever was causing the extraneous metadata you noticed on some records, we should decide in what contexts to have the software delete existing metadata, implement and test that deletion functionality, and then run another bulk operation to remove this remaining metadata from each of the records. It seems like that will all need to happen before the MS0020 update can be considered complete.

**Comment by rmjaffe** [comment]
"It seems like that will all need to happen before the MS0020 update can be considered complete." I think this is something we would need to discuss.  The pieces of metadata that I'd like removed (subseries, theme) aren't doing any great harm by being there and we don't have to share them out to Calisphere or Primo. If it makes sense to do the work around deletion now, great.  If not, I think we leave it outside of what constitutes "done" for this phase of work and address it later.

**Comment by NedHenry** [comment]
In regard to the reported issue with extra subject headings on this work: I checked the ingest/update spreadsheet,. The extra subject headings for "Baby D" and "Al Lowry" are indeed present in the spreadsheet, intentionally or otherwise.

**Comment by NedHenry** [comment]
For the pre-existing metadata that should have been removed, I think I should be able to write a script to remove that metadata for this collection relatively easily after I resolve the rest of the restructuring ingest. I know we talked about this metadata-wiping functionality for the bulk updates in the past, but between all the other complications with complex objects I never remembered to implement it. 
For the future, I want to add an extra option for each bulk update to determine whether or not to remove all existing metadata from records before adding/updating the metadata from the spreadsheet. I think the default should be to preserve the existing metadata, so that nobody accidentally deletes data they expect to retain. However, in cases like this, we would want to be able to set the bulk update to delete all metadata from each object before adding the metadata from the spreadsheet. I will create a ticket for that, which we should resolve before our next bulk update.

**Comment by NedHenry** [comment]
Also, I have a theory about what is going on with the broken links between child and parent records. I believe it is related to the fedora performance problems (#160) that are holding up this ingest. I will explain in more detail on that ticket what is going on. Once I resolve this in the bulk operations software, I should be able to fix these structural issues en masse.

**Comment by rmjaffe** [comment]
@NedHenry, that's good news about the errant subject terms.  I'm glad it's a spreadsheet snafu instead of an ingest issue.  Thanks for checking into it.  I'm going to look into where those subject terms slipped in because they weren't on the metadata sheet I was consulting.

In reading through your comments, I'm remembering that we never completely nailed down functional requirements for the bulk update, so it would be helpful, at least to me, to meet at some point to talk more about the metadata overlay rules/setup. I can make a ticket for this just to bookmark it.

I agree, keeping the pre-existing metadata guards against nuking anything we didn't want to lose, but I'm certain there will be more cases where we'll want to be able to remove certain properties from the records. Perhaps the solution is to create a way to indicate "remove this metadata property" on the ingest spreadsheets?

So I'm clearer on how the bulk updates are working now -- pre-existing metadata in pre-existing properties are not being removed, new metadata is being added, where there are changes to pre-existing metadata the older values are being replaced with the newer values?

**Comment by kellyChex** [comment]
Two things:
1. Click on the image on the browser results you get a message that says the object doesn't exist 
2. The metadata and the object don't match. 

@chesleyperry To create new tickets at top of backlog with screenshots and links
